internal/authproxy/serve.go

```diff
@@ -132,6 +144,37 @@ func Command() (*cobra.Command, error) {
 	return serveCmd, nil
 }
 
+type (
+	RequestStats struct {
+		LastRequest time.Time
+		mutex       sync.RWMutex
+	}
+)
+
+func NewStats() *RequestStats {
+	return &RequestStats{
+		LastRequest: time.Now(),
+	}
+}
+
+func (l *RequestStats) Process(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+
+		if err := next(c); err != nil {
+			c.Error(err)
+		}
+		l.mutex.Lock()
+		defer l.mutex.Unlock()
+		l.LastRequest = time.Now()
+		return nil
+	}
+}
+func (l *RequestStats) Handle(c echo.Context) error {
+	l.mutex.RLock()
+	defer l.mutex.RUnlock()
+	return c.String(http.StatusOK, l.LastRequest.Format("2006-01-02 15:04:05"))
```

Comment:
suggestion: I know we briefly talked about this and I originally agreed to this. But after thinking about it, it is better to make this a json dict. That way we can update this in a backwards compatible way by adding different fields and keeping the old fields around. Whereas with just a string like this pretty much any change we make will be a breaking one. And we want this to be robust because a new version of amalthea will be calling the endpoints on old sessions that are just hanging around. For now it will have 1 single key but that is ok. You can make it a `map[string]string` and serialize that - you dont need a whole new struct just for this.

Reply:
Also lets use a stndard for the datetime format. In go you have builtin `time.RFC3339` which you can pass straight to `.Format()`